openadapt/ml_models/openai_module/gpt3_davinci/finetune.py

```diff
@@ -0,0 +1,79 @@
+from openadapt import models
+from openadapt.crud import get_action_events, get_window_events
+import fire
+import openai
+from copy import deepcopy
+import utils
+
```

Comment:
Should we have a generic `ModelFineTuner` like this

```python
class ModelFineTuner(ABC):
    def __init__(self, model_name: str):
        self.model_name = model_name

    @abstractmethod
    def get_finetune_pairs(self, recording: Recording):
        pass
    
    @abstractmethod
    def sanitize_prompt(self, prompt):
        pass

    @abstractmethod
    def tune_model(self, prompt_dataset: Any):
        pass

```

And here in `gpt3_davinci`, you can inherit from `ModelFinetuner` and keep the rest of your code.

It would helps a lot for the following folks who need to fine-tune another model and keep our fine-tune pipeline really organised and structured.

WDYT ?